cpp/src/arrow/compute/kernels/scalar_cast_internal.h

```diff
@@ -43,6 +43,7 @@ struct CastFunctor<
 };
 
 Status CastFromExtension(KernelContext* ctx, const ExecSpan& batch, ExecResult* out);
+Status CastToExtension(KernelContext* ctx, const ExecSpan& batch, ExecResult* out);
```

Comment:
I'm not sure it's useful to expose this function as it's not called directly anywhere.